src/app/cli/src/cli_commands.rs

```diff
@@ -546,3 +540,25 @@ where
 
     Ok(result_refs)
 }
+
+fn parse_server_url(maybe_url_str: Option<&String>) -> Result<Option<Url>, CLIError> {
```

Comment:
Can we move this function on a `cli_parser.rs` level (similarly to `value_parse_dataset_ref_local`), to fail argument parsing if URLs are invalid, and to extract already parsed `Url` types instead of raw strings.

Reply:
Moved